### PR TITLE
Viewing single past order

### PR DESCRIPTION
**Pull request by sabondano**
This PR implements every aspect of user story #11 except the part that says:

> And if any of the products in the order were retired from the menu
Then they should still be able to access the item page
But they should not be able to add the item to their cart

It looks like this should be part of a different user story once we implement a feature that allows admins to retire products from the menu.

What do you all think? Other than that let me know if you have any comments.

### CODE REVIEW

**Comment by sabondano** [comment]
I also wanted to show you what the order show view looks like as of now, incase you want to change anything.

FYI, the well that says "other notes" is only displayed when an order has a status of "cancelled" or "completed."

screen shot 2015-08-09 at 8 21 01 pm

**Comment by mdorrance** [comment]
I like the view. Thanks for sharing.

**Comment by sabondano** [comment]
:+1:

File: spec/features/user_sees_single_past_order_spec.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
+require "rails_helper"
+
+feature "user sees a single past order" do
+  context "as an authenticated user with one previous order" do
+    before do
+      user = User.create(first_name: "Jane",
+                         last_name:  "Doe",
+                         email:      "jane@gmail.com",
+                         password:   "password")
```
Comment (imwithsam) [inline spec/features/user_sees_single_past_order_spec.rb]: Look like this is just pretty formatting. No need to change.

File: spec/features/user_sees_single_past_order_spec.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
+require "rails_helper"
+
+feature "user sees a single past order" do
+  context "as an authenticated user with one previous order" do
+    before do
+      user = User.create(first_name: "Jane",
+                         last_name:  "Doe",
+                         email:      "jane@gmail.com",
```
Comment (imwithsam) [inline spec/features/user_sees_single_past_order_spec.rb]: Look like this is just pretty formatting. No need to change.

File: spec/features/user_sees_single_past_order_spec.rb [hunk 3]
```diff
@@ -0,0 +1,80 @@
+require "rails_helper"
+
+feature "user sees a single past order" do
+  context "as an authenticated user with one previous order" do
+    before do
+      user = User.create(first_name: "Jane",
+                         last_name:  "Doe",
```
Comment (imwithsam) [inline spec/features/user_sees_single_past_order_spec.rb]: Look like this is just pretty formatting. No need to change.

File: spec/features/user_sees_single_past_order_spec.rb [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                       quantity: 1,
+                       unit_price: @product_1.price)
+
+      OrderItem.create(order_id: @order.id,
+                       product_id: @product_2.id,
+                       quantity: 2,
+                       unit_price: @product_2.price)
+
+      allow_any_instance_of(ApplicationController)
+        .to receive(:current_user).and_return(user)
```
Comment (imwithsam) [inline spec/features/user_sees_single_past_order_spec.rb]: Look like this is just pretty formatting. No need to change.

File: spec/features/user_sees_single_past_order_spec.rb [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      expect(page).to have_content("$49.97") # total
+    end
+
+    scenario "if order was completed/cancelled, timestamp should be displayed" do
+      @order.update(status: "cancelled")
+      @order.update(updated_at: DateTime.civil(2015, 8, 2, 21, 33, 0))
+
+      visit order_path(@order)
+      within(".other-notes") do
+        expect(page).to have_content("This order was cancelled on August  2, 2015 at 9:33 PM.")
```
Comment (imwithsam) [inline spec/features/user_sees_single_past_order_spec.rb]: We can probably split this up on the . or by breaking the string with a \ concatenator. Also, is there meant to be an extra space before the 2 in August 2?

File: spec/features/user_sees_single_past_order_spec.rb [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      expect(current_path).to eq(orders_path)
+
+      within("td.order-details-link") do
+        click_link("View Order Details")
+      end
+
+      expect(current_path).to eq(order_path(@order))
+
+      expect(page).to have_content("Order ID: # #{@order.id}")
+      expect(page).to have_content("Purchase Date/Time: July  5, 2015 at  9:33 PM")
```
Comment (imwithsam) [inline spec/features/user_sees_single_past_order_spec.rb]: Is there meant to be a space before the 5 in July 5?

File: app/models/order.rb [hunk 7]
```diff
@@ -1,6 +1,10 @@
 class Order < ActiveRecord::Base
   belongs_to :user
   has_many :order_items
+  has_many :products, through: :order_items
+
+  enum status: ["ordered", "paid", "cancelled", "completed"]
+  validates :status, inclusion: { in: ["ordered", "paid", "cancelled", "completed"] }
```
Comment (imwithsam) [inline app/models/order.rb]: Break this line after the first , ?

File: spec/features/user_sees_single_past_order_spec.rb (hunk 1 above)
Comment (sabondano) [inline spec/features/user_sees_single_past_order_spec.rb]: On the same page.

File: spec/features/user_sees_single_past_order_spec.rb (hunk 2 above)
Comment (sabondano) [inline spec/features/user_sees_single_past_order_spec.rb]: On the same page.

File: spec/features/user_sees_single_past_order_spec.rb (hunk 3 above)
Comment (sabondano) [inline spec/features/user_sees_single_past_order_spec.rb]: On the same page.

File: spec/features/user_sees_single_past_order_spec.rb (hunk 4 above)
Comment (sabondano) [inline spec/features/user_sees_single_past_order_spec.rb]: On the same page.

File: spec/features/user_sees_single_past_order_spec.rb (hunk 5 above)
Comment (sabondano) [inline spec/features/user_sees_single_past_order_spec.rb]: Yes, strftime adds the space before the 1 digit date values. Will split per your recommendation.

File: spec/features/user_sees_single_past_order_spec.rb (hunk 6 above)
Comment (sabondano) [inline spec/features/user_sees_single_past_order_spec.rb]: strftime adds it for single digit date values.

File: app/models/order.rb (hunk 7 above)
Comment (sabondano) [inline app/models/order.rb]: Will do it.

**Comment by mdorrance** [comment]
@imwithsam and I agree on merge.

**Pull request merged**